test/integration/backward_compatible/serial/ClientReconnectTest.js

```diff
@@ -59,6 +40,37 @@ describe('ClientReconnectTest', function () {
         await testFactory.shutdownAll();
     });
 
+    it('should send the client state to the cluster after reconnections, ' +
+        +'regardless it is connected back to possibly the same cluster with the same id or not.', async function () {
+        const fakeInitializeClientOnCluster = sandbox.replace(
```

Comment:
Just a note: This test will probably fail in backward compatibility tests as there is no client version filter

Reply:
@harunalpak I overlooked this, can you add a version check?